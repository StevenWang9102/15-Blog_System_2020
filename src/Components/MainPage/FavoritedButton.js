import React from "react";
import { useState } from "react";
import { createUseStyles } from "react-jss";
import PropTypes from "prop-types";

const useStyles = createUseStyles({
  myButton: {
    textAlign: "left",
    marginLeft: "10px",
    "&:hover": {
      color: "white",
      fontWeight: "500",
      textAlign: "center",
      backgroundColor: " #5cb85c94"
    },
    "&:focus": {
      color: "#5cb85c",
      fontWeight: "500",
      textAlign: "center",
      backgroundColor: "#5cb85c11"
    }
  }
});

export const FavoritedButton = props => {
  const [favoritesCountPlus, setFavoritesCountPlus] = useState(0);
  const Button = ({ children }) => {
    const classes = useStyles();
    return (
      <button
        type='button'
        className={`btn btn-outline-primary btn-sm pull-xs-right ${classes.myButton}`}
        onClick={() => {
          
          // Switch method between "POST" and "DELETE"
          if (props.userInformation.token) {
            const tempMethod = { ...props.httpMethod };

            if (tempMethod[props.article.slug] === "POST") {
              tempMethod[props.article.slug] = "DELETE";
            } else {
              tempMethod[props.article.slug] = "POST";
            }

            props.onFavoritedArticleClicked(
              props.userInformation.token,
              props.article.slug,
              tempMethod[props.article.slug],
              props.currentPageOffSet,
            );
            props.setHttpMethod(tempMethod);

            // Display add 1 or not
            setFavoritesCountPlus(
              tempMethod[props.article.slug] === "POST" ? 1 : 0
            );
          }
        }}>
        {children}
      </button>
    );
  };

  return (
    <Button>
      <img src='../icon/002-heart-2.png' alt='' />
      {props.article.favoritesCount + favoritesCountPlus}
    </Button>
  );
};


FavoritedButton.propTypes = {
  userInformation: PropTypes.object,
  article: PropTypes.object,
  onFavoritedArticleClicked: PropTypes.func,
  setHttpMethod: PropTypes.func,
};