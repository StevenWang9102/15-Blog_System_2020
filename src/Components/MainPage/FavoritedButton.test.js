import React from "react";
import { FavoritedButton } from "./FavoritedButton";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <FavoritedButton
        userInformation={{}}
        article={{}}
        onFavoritedArticleClicked={() => {}}
        setHttpMethod={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
