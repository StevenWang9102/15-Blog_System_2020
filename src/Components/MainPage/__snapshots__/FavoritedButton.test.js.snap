// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn btn-outline-primary btn-sm pull-xs-right myButton-0-2-1"
  onClick={[Function]}
  type="button"
>
  <img
    alt=""
    src="../icon/002-heart-2.png"
  />
  NaN
</button>
`;
